src/main/java/frc/robot/subsystems/BallTransport.java

```diff
@@ -0,0 +1,118 @@
+package frc.robot.subsystems;
+
+import java.util.function.BooleanSupplier;
+import com.chopshop166.chopshoplib.SampleBuffer;
+import com.chopshop166.chopshoplib.commands.SmartSubsystemBase;
+import com.chopshop166.chopshoplib.motors.SmartMotorController;
+import com.chopshop166.chopshoplib.sensors.IColorSensor;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.maps.RobotMap.BallTransportMap;
+
+public class BallTransport extends SmartSubsystemBase {
+    private final SmartMotorController bottomMotor;
+    private final SmartMotorController topMotor;
+    private final IColorSensor colorSensor;
+    private final BooleanSupplier laserSwitch;
+    private final Alliance allianceColor;
+
+    private boolean cargoInColorSensor = false;
+    private final double SHUNT_SPEED = 1.0;
+    private final double TRANSPORT_SPEED = 1.0;
+
+    // creates a color sample buffer.
+    private SampleBuffer<Color> colorBuffer = new SampleBuffer<Color>(2);
```

Comment:
Pro tip - you can actually remove the generic specialization on the right hand side

```java
private SampleBuffer<Color> colorBuffer = new SampleBuffer<>(2);
```

Reply:
This was fixed